src/types/input.jl

```diff
@@ -1,23 +1,29 @@
-### Inputs ###
-
 abstract type InputType end
 
-struct Bosonic <:InputType
+struct Bosonic <: InputType
 end
-struct PartDist <:InputType
+
+abstract type PartDist <: InputType end
+struct OneParameterInterpolation <: PartDist
 end
-abstract type ToyModel <: InputType
+struct RandomGramMatrix <: PartDist
 end
-abstract type RandomModel <: InputType
+struct UserDefinedGramMatrix <: PartDist
 end
-struct Distinguishable <:InputType
+
+struct Distinguishable <: InputType
 end
-struct Undef <:InputType
+
+struct Undef <: InputType
 end
 
+abstract type Gaussian <: InputType end
```

Comment:
unless you use this I would remove it - or otherwise let's do supertypes FockInput and GaussianInput etc